sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp

```diff
@@ -1,5 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
+#if !defined(INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP)
```

Comment:
why do we need include guard - isn't `#pragma once` enough? We use `#pragma once` everywhere, we decided to use it and not include guards.

Reply:
[C++ Core Guidelines SF.8](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-guards) says that `#pragma` is a non-standard extension and should be avoided in favor of using the language standard.

If the Azure C++ guidelines state that we're intentionally ignoring SF.8, then we can remove it, but otherwise IMHO we should follow the best practices of the language.

Reply:
I had the same argument, but when deciding on guidelines, we decided to not follow strictly standard for `#pragma once`.

Reply:
I had the same argument, but when deciding on guidelines, we decided to not follow strictly standard for `#pragma once`.

Reply:
https://azure.github.io/azure-sdk/cpp_introduction.html#cpp-style-headerguards - it does not explicitly say "do not use include guards" [but use pragma once instead], but I remember that was the intent. We can clarify during our Tuesday meeting.

Reply:
https://azure.github.io/azure-sdk/cpp_introduction.html#cpp-style-headerguards - it does not explicitly say "do not use include guards" [but use pragma once instead], but I remember that was the intent. We can clarify during our Tuesday meeting.